### PR TITLE
initio: Add safety checks inside MraaIo destructor

### PR DESCRIPTION
**Pull request by stefan-andritoiu**
Signed-off-by: Adelin Dobre <adelin.dobre@rinftech.com>

### CODE REVIEW

File: api/mraa/initio.hpp [hunk 1]
```diff
@@ -109,39 +109,41 @@ class MraaIo
...
-        if (descs->n_aio) {
-            free(descs->aios);
-        }
-        if (descs->n_gpio) {
-            free(descs->gpios);
-        }
-        if (descs->n_i2c) {
-            free(descs->i2cs);
-        }
+        if(descs != nullptr) {
```
Comment (alext-mkrs) [inline api/mraa/initio.hpp]: This needs a space after `if`

File: src/initio/initio.c [hunk 2]
```diff
@@ -787,7 +787,6 @@ mraa_io_init(const char* strdesc, mraa_io_descriptor** desc)
                     strncat(new_desc->leftover_str, str_descs[i], strlen(str_descs[i]));
                 }
 
-                strncat(new_desc->leftover_str, str_descs[i], strlen(str_descs[i]));
```
Comment (alext-mkrs) [inline src/initio/initio.c]: How is this related to "Add safety checks inside MraaIo destructor"? Although it indeed looks strange given we do the same right above in the `else` branch, I suggest splitting this out to a separate commit to avoid polluting git history by commits where unrelated stuff is crammed together.

**Comment by stefan-andritoiu** [comment]
Split the commit into two separate commits
@alext-mkrs , can you take another look?

**Comment by alext-mkrs** [comment]
Sorry for the delay - this one fell through the cracks. Looks good to me now!

**Comment by alext-mkrs** [comment]
@Propanu, just as a precaution, do you see any problems with merging this one, in view of the upcoming repo transition?

**Comment by Propanu** [comment]
@alext-mkrs, not at all, in fact we can continue merging as usual while Eclipse will review an initial repo snapshot. Actually, I plan to start preparing the UPM 2.0 release while the projects are under IP review with contributions from RINF. The only kind of merge that would raise a flag would be something that changes the inbound license table, and this is definitely not the case.

**Comment by alext-mkrs** [comment]
Got it, thanks - I'll then merge this sometime this week, maybe the next weekend when I got time.

**Comment by tingleby** [comment]
Merged, Thanks!

**Comment by alext-mkrs** [comment]
Thanks Tom :)
